core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -1693,6 +1696,16 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
     assert(locs(blockIds(0)) == expectedLocs)
   }
 
+  test("SPARK-30594: Do not post SparkListenerBlockUpdated when updateBlockInfo returns false") {
+    // update block info for non-existent block manager
+    val updateInfo = UpdateBlockInfo(BlockManagerId("1", "host1", 100),
+      BlockId("test_1"), StorageLevel.MEMORY_ONLY, 1, 1)
+    val result = master.driverEndpoint.askSync[Boolean](updateInfo)
```

Comment:
how do we make sure this request returns false?

Reply:
Because the `BlockManagerId("1", "host1", 100)` hasn't registered at driver. And driver will response `false` for the block from a non-existent block manager.

Reply:
Yep, exactly as Yi said.